packages/bierzo-wallet/src/store/usernames/selectors.ts

```diff
@@ -0,0 +1,5 @@
+import { RootState } from '../reducers';
+import { BwUsername } from '.';
+
+export const getUsername = (state: RootState): BwUsername | undefined =>
+  Object.values(state.usernames).find(username => username);
```

Comment:
What is the reasoning behind this? Get the first username? Why? Are we assuming they are going to be the same for all addresses?

Can you add a comment with some explanations?

Reply:
1. This is not possible to show different usernames according to current wallet spec.
2. I saw this code snippet in `wallet-demo`: 
```
const bnsConn = value.connection as BnsConnection; dispatch(setBnsChainId(bnsConn.chainId()));
```
which shows me that `bnsConn.chainId()` contains bns chain id.
And I decided as we are checking for Bns in [/usernames/actions.ts#L17](https://github.com/iov-one/ponferrada/blob/85e6ed75847d4282684e5d7f4b63706166c3f755/packages/bierzo-wallet/src/store/usernames/actions.ts#L17)  and according to the whole of this method logic then I can use first record from `usernames` state as username of the current logged in user.

Correct me if I am wrong. @webmaster128.

Reply:
Logic is okay for now. Multiple usernames are an edge case we can take care of later on.

But please update code a little make explicit this is a "first" getter. By the way, the simpler way to implement "first§ in a type-safe way is `.find(() => true)`.

```ts
export const getUsername = (state: RootState): BwUsername | undefined => {
  const firstUsername = Object.values(state.usernames).find(() => true);
  return firstUsername;
}
  
```

Reply:
> This is not possible to show different usernames according to current wallet spec.

I am not saying you need to support multiple usernames (one by chain), but if we just take the first one, due a product restriction, we should comment it. For avoid confusion.

If you take a look into the usernames reducer you will see an array of usernames, one per key. In the wallet demo we had the same username for all chains, but unless I am wrong we could have one per chain.

My point is, seeing the username is tied to a chain id, having a `getUsername` selector without having specified somehow the chainId is kind of incorrect. I suggest:
```
const getUsernameByChainId = createSelector(
  state => Object.values(state.usernames),
  (_, chainId) => chainId,
  (usernames, chainId) => usernmaes.find(username => username.chainId === chainId)
)

....
// in react component
const username = useSelector(state =>
    getUsernameByChainId(state, 'bnsChainId')
  ))
```
👀 👀 👀 
For now, I suggest to hardcode the bnsChainId, but contact @webmaster128 and ask **him to create a ticket** for introducing in the extension reducer an action for spec the bns chain id as separate property:
```
export interface ExtensionState {
  readonly connected: boolean;
  readonly installed: boolean;
  readonly keys: { [chain: string]: string };
  readonly bnsChainId: string;
}
```
Or if you think this is messing up things, just create `src/store/chains` reducer and place there the chainIds of supported chains.

Reply:
@webmaster128 can you validate the reasoning here and then, @albertandrejev create a ticket pitching Martin for validating it?

Reply:
> My point is, seeing the username is tied to a chain id, having a getUsername selector without having specified somehow the chainId is kind of incorrect.

Usernames only exist ion the BNS chain. What we want to show is all usernames that the user owns (typically 0 or 1). I will go though the code that creates this data today and try to clarify the situation. Albert's use makes sense to me.

Reply:
Yes, this was my first idea is to hardcode chain id.

Reply:
@apanizo #408.